app/models/relationship.rb

```diff
@@ -0,0 +1,7 @@
+class Relationship < ActiveRecord::Base
+  belongs_to :follower, class_name: "User"
+  belongs_to :leader, class_name: "User"
+
+  validates_presence_of :leader_id, :follower_id
+  validates_uniqueness_of :leader_id, scope: :follower_id
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.